engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/NadelCoerceTransform.kt

```diff
@@ -128,7 +140,18 @@ internal class NadelCoerceTransform : NadelTransform<State> {
             result,
             overallField,
         ) { value ->
-            state.fieldType.coercing.parseValue(value)
+            coerceValue(state, value)
+        }
+    }
+
+    private fun coerceValue(state: State, value: Any): Any {
+        val coercedValue = state.fieldType.coercing.parseValue(value)
+        if (coercedValue != value) {
```

Comment:
Just FYI, it might've been worth to have just checked the javaClass instead of this. It's a scalar, but it could be a _HUGE_ JSON blob. I've definitely seen those returned by services.